sdk/identity/azure-identity/src/managed_identity_source.cpp

```diff
@@ -81,17 +83,26 @@ Azure::Core::Credentials::AccessToken AppServiceManagedIdentitySource::GetToken(
     Azure::Core::Credentials::TokenRequestContext const& tokenRequestContext,
     Azure::Core::Context const& context) const
 {
-  return TokenCredentialImpl::GetToken(context, [&]() {
-    auto request = std::make_unique<TokenRequest>(m_request);
+  std::string scopesStr;
+  {
+    auto const& scopes = tokenRequestContext.Scopes;
+    if (!scopes.empty())
     {
-      auto const& scopes = tokenRequestContext.Scopes;
-      if (!scopes.empty())
+      scopesStr = TokenCredentialImpl::FormatScopes(scopes, true);
+    }
+  }
+
+  return TokenCache::GetToken(std::string(), GetClientId(), GetAuthorityHost(), scopesStr, [&]() {
+    return TokenCredentialImpl::GetToken(context, [&]() {
```

Comment:
What is being captured with the inner universal byref capture? It doesn't look like there's anything other than scopestr, so it might be better to restrict the capture to just scopestr.

Reply:
It is `[this, &context, &scopesStr]`. I can add it, but see my comment below. If you think we still should, let me know and I'll add it.

Reply:
In general, I prefer smaller lambda captures. But it's not a huge deal, especially if you know that the function is synchronous.